### DISCUSSION TITLE
RenderTarget with Shared Context

### DISCUSSION THREAD

**Discussion by jwinarske** [google/filament#4475] [Q&A]
Android can access the Texture ID indirectly via SurfaceTexture.  How to access it in Linux?

Do I simply create Texture prior to initializing Filament, then call `Texture.Builder().importTexture()` with the desired texture id?

I'm running samples/rendertarget.cpp w/shared context on Linux.

**Comment by romainguy** [comment]
@pixelflinger

**Comment by jwinarske** [comment]
I'm thinking I just use the values of app.offscreenColorTexture and app.offscreenDepthTexture with my primary context.
https://github.com/google/filament/blob/main/samples/rendertarget.cpp#L205
https://github.com/google/filament/blob/main/samples/rendertarget.cpp#L209

Plus exposing an API "getId" to return the actual Id.

**Reply by dandingol03** [reply]
Hi, how is your progress. I also need to render stuff to SurfaceTexture.

**Reply by jays0606** [reply]
Hi! How were you able to access textureID from surfaceTexture in android ?! 
I've looked really hard but could not find a way !

**Comment by jwinarske** [comment]
I'm rendering Filament to a Wayland Vulkan Surface, and using HW Alpha Blending to blend with UI HW layer.  This subject is no longer any interest to me.
